tools/deployment_tools.js

```diff
@@ -0,0 +1,180 @@
+const web3 = require('web3');
+
+class ContractRegistry {
+  constructor() {
+    this.startAddress = '0x0000000000000000000000000000000000000100';
+    this.nextAvailableAddress = web3.utils.hexToBytes(this.startAddress);
+    this.contracts = {};
+    this.instances = [];
+
+    this.addContract = this.addContract.bind(this);
+    this.addInstance = this.addInstance.bind(this);
+    this.checkFullyLinked = this.checkFullyLinked.bind(this);
+    this.link = this.link.bind(this);
+    this.linkBytecode = this.linkBytecode.bind(this);
+    this.linkedBytecodeForContract = this.linkedBytecodeForContract.bind(this);
+    this.truffleDeployer = this.truffleDeployer.bind(this);
+    this.toParityGenesisAccounts = this.toParityGenesisAccounts.bind(this);
+    this.takeNextAddress = this.takeNextAddress.bind(this);
+  }
+
+  takeNextAddress() {
+    const addressHex = web3.utils.bytesToHex(this.nextAvailableAddress);
+    this.nextAvailableAddress[this.nextAvailableAddress.length - 1] += 1;
```

Comment:
Wouldn't it be easier to have a `BN` or similar and actually increase that? If I understand this approach correctly it is bound to 10 addresses, which may or may not be sufficient in the future.

Reply:
It's bound to 255 (as it increments the last byte, not string character), but yeah, probably possible to use `BN` instead. I'll look into that.

Reply:
Doesn't it increment the last hex character (4 bits => max 16)? And does it know to go from `9` to `a`? JavaScript magic? 🤔

Reply:
`this.nextAvailableAddress` is a byte array, so no magic there 😉

Reply:
But that code is gone now anyway 😄

Reply:
Ah, somehow I thought it was a hex string 😳